### ISSUE TITLE
Deprecate withMonitor or fuse withMonitorRef

### ISSUE THREAD

**Issue by mboes** [haskell-distributed/distributed-process#281]
I think it's confusing to have both `withMonitor` and `withMonitorRef`. The two functions are nearly identical. Could we either deprecate `withMonitor`, since `withMonitorRef` is all that's needed, or (at the cost of a breaking change) make the naming consistent with base: i.e.

```haskell
withMonitor :: ProcessId -> (MonitorRef -> Process a) -> Process a
withMonitor_ :: ProcessId -> Process a -> Process a
```

(as in `mask_` and `bracket_` in `Control.Exception` and elsewhere)

If we deprecate `withMonitor` now, it leaves us free to recycle that name to make the naming consistent in a later release.

**Comment by hyperthunk** [comment]
+1

**Comment by qnikst** [comment]
I agree to make such patch, but I'm not sure what will our strategy be here, and as consequence what to put into deprecation message.

Will `DEPRECATED withMonitor "this function is a subject of refactoring in next major release see issue #281 for details"` be ok? As it seems that such message will cover all possible cases.

**Comment by hyperthunk** [comment]
I think that will be fine. @mboes?

I also think we need to start creating branches for future releases that contain foxes we don't want to merge immediately into master. The release process then becomes, rebase master onto the release-x.y.z branch, make a PR for the release branch onto master, retest and review, merge the release PR and tag master.
Thoughts? @mboes, facundo? @qnikst

The alternative is to have PRs standing that we don't want in a patch release and having to remember not to merge them. When they're made against a release branch, it's obvious they don't belong in master straight away, the only downside being situations like this in which there are potential merge conflicts when we rebase master and the release branch. But in those cases and this one, we can take the temporary fix patch to master and the release branch (or merge master into it immediately if there aren't too many diverging points in the code base), then apply the second (API breaking) fix. This also gives downstream libraries using stack, the opportunity to test against the planned API breaking release ahead of time, but managing their own branches in a similar fashion.

**Comment by hyperthunk** [comment]
Oh and we can up the version number in cabal immediately on the release branch, which gives downstreams with a tight upper bound - way to avoid building against breaking changes without realising.

**Comment by hyperthunk** [comment]
I've deleted my comments/questions about the branching process. We can discuss it on the mailing list instead.

@qnikst that deprecation message looks fine to me.

**Comment by hyperthunk** [comment]
I'm going to pull together a fix for the tracing bugs and a hope to close out a number of other open issues, so let's sit tight until I've got enough changes for a major release.

**Comment by qnikst** [comment]
Ok, then we should verify that everything is in master and provide deprecation message, will you do that yourself or want me to take care of that.

**Comment by hyperthunk** [comment]
I'll do it once the rest of my patches have been merged.
